### ISSUE TITLE
Unraid: migrating from docker-compose to docker-immich

### ISSUE THREAD

**Issue by vulubalulu** [imagegenius/docker-immich#12]
I have do have a immich instance running with docker-compose. How would I go about migrating all my data to the docker version. Simply copy pasting the upload folder did not do the trick and i seem not to be able to locate the actual docker-files from docker-compose except for the compose and the env. file. I am not techsavvy in this regard, looking for guidance. thank you.

**Comment by martabal** [comment]
Hi !
You just have to use the same upload folder for `/photos` and use the previous PostgreSQL database. 

Note that you need to create a PostgreSQL 14 and a Redis container. Use the same appdata folder for PostgreSQL or you can migrate the database from the previous PostgreSQL to the new one.

**Comment by hydazz** [comment]
how did you go with this? where you able to get it working?

**Comment by vulubalulu** [comment]
Thanks for the quick reply. I managed to locate the database data (it was in /var/lib/docker/volumes/immich_pgdata) and copy paste it into the postgresql14 docker container. The immich container starts up but i get the same error message as https://github.com/imagegenius/docker-immich/issues/13.
`is the immich and postgres container on the same docker network? or are they able to communicate with each other?`
Yes they are on the same network. 

that's what the log says

![image](https://user-images.githubusercontent.com/62702768/214148608-dd359c3b-ba50-4c96-96b0-9138406d2e86.png)

**Comment by vulubalulu** [comment]
docker run:
![image](https://user-images.githubusercontent.com/62702768/214150676-42d93a37-88f4-4f35-ab9f-a0a857256833.png)

**Comment by hydazz** [comment]
@martabal, i dont think that we will support a 'direct' migration from the immich docker compose.

@vulubalulu, if its not too much fuss you can try the below:

1. stop both immich and postgres containers
2. move the immich photos folder somewhere safe (`/mnt/cache/immich` in your case) `mv /mnt/cache/immich /mnt/cache/immich_backup)`
3. delete the immich and postgres appdata and setup immich and postgres from scratch using the CA template (don't delete the photos folder!)
4. sign into immich (if it works), click on the profile icon on the top right, > manage your immich account > API keys > new key (copy)
5. run `cd /mnt/cache/immich_backup`
6. run `docker run -it --rm -v '$(pwd)':/import ghcr.io/immich-app/immich-cli:latest`
7. run `immich upload --key <copied API key> --server http://<IP>:2283/api -d /uploads`

**Comment by hydazz** [comment]
@martabal if this works can you open a pr to add this to the readme please :)

**Comment by martabal** [comment]
Technically, it works, I've already done it.
But it would be much simpler to just have a script to run.



> @martabal if this works can you open a pr to add this to the readme please :)

Yep !

**Comment by vulubalulu** [comment]
> @martabal, i dont think that we will support a 'direct' migration from the immich docker compose.
> 
> @vulubalulu, if its not too much fuss you can try the below:
> 
>     1. stop both immich and postgres containers
> 
>     2. move the immich photos folder somewhere safe (`/mnt/cache/immich` in your case) `mv /mnt/cache/immich /mnt/cache/immich_backup)`
> 
>     3. delete the immich and postgres appdata and setup immich and postgres from scratch using the CA template (don't delete the photos folder!)
> 
>     4. sign into immich (if it works), click on the profile icon on the top right, > manage your immich account > API keys > new key (copy)
> 
>     5. run `cd /mnt/cache/immich_backup`
> 
>     6. run `docker run -it --rm -v '$(pwd)':/import ghcr.io/immich-app/immich-cli:latest`
> 
>     7. run `immich upload --key <copied API key> --server http://<IP>:2283/api -d /uploads`

The steps did not work for me because step 6 threw errors at me, something about absolute paths, i tried to troubleshoot it but to no avail. Instead through hours of trial and error i found the reason for my issue.

1.  photo share container path
 i was not pointing to the upload folder itself but a level above that's what caused the error
❌  `/mnt/cache/immich` 
✅ `/mnt/cache/immich/uploads`

2. i had permission error with the following path `/app/immich/server/.reverse-geocoding-dump/`
![image](https://user-images.githubusercontent.com/62702768/214177076-2f9a451b-fc2e-49ed-a8f4-6617bc8fbc05.png)
`chmod -r 777 /app/immich/server/` solved the error, but i do not believe that this was an issue.

**to sum it up** @martabal was correct with his suggestion to use the same upload folder and postgresql database.
after creating* postgresql14 docker delete the folder's content and replace it with the content of `/var/lib/docker/volume/<immich_docker_compose_name>/_data/`.
*When creating the postgresql docker, make sure to use the same database name, username and password
Then point your photo share

**Comment by hydazz** [comment]
> chmod -r 777 /app/immich/server/ solved the error

`chown -R abc:abc app/immich/server/.reverse-geocoding-dump`*

Instructions for me to do something with in the future:

**docker-compose migration**:

move the contents of `/var/lib/docker/volume/<immich_docker_compose_name>/_data/` to `/mnt/user/appdata/postgres14`
start the postgres 14 container and mount it to `/mnt/user/appdata/postgres14` on the host.

**remove all variables from the postgres template: `POSTGRES_USER`, `POSTGRES_PASSWORD`, `POSTGRES_DB`**
the database is already initialised and these variables dont do anything.

run `mv /mnt/user/immich/uploads /mnt/cache/immich-migrate` 
then `mv /mnt/cache/immich-migrate /mnt/user/<wherever you desire>`

`/mnt/user/<wherever you desire>` can be mounted to /photos within the container

@vulubalulu have I left anything out?, we will use this to add instructions to the readme

**Comment by hydazz** [comment]
>i had permission error with the following path /app/immich/server/.reverse-geocoding-dump/

fixed in #14

**Comment by vulubalulu** [comment]
@hydazz I believe you have covered it all. it's good to go.

**Comment by hydazz** [comment]
closing this as a solution to the **main** issue here has been added to the readme: https://github.com/imagegenius/docker-immich#unraid-migrate-from-docker-compose
